pkg/registry/resource/resourceclaim/strategy.go

```diff
@@ -119,7 +136,10 @@ type resourceclaimStatusStrategy struct {
 	resourceclaimStrategy
 }
 
-var StatusStrategy = resourceclaimStatusStrategy{Strategy}
+// NewStatusStrategy creates a strategy for operating the status object.
+func NewStatusStrategy(resourceclaimStrategy *resourceclaimStrategy) *resourceclaimStatusStrategy {
+	return &resourceclaimStatusStrategy{*resourceclaimStrategy}
+}
 
 // GetResetFields returns the set of fields that get reset by the strategy and
 // should not be modified by the user. For a status update that is the spec.
```

Comment:
Update all of these receivers on `resourceclaimStatusStrategy` to be pointers as well.